osbuild/__init__.py

```diff
@@ -69,13 +70,156 @@ def __exit__(self, exc_type, exc_value, exc_tb):
         self.root = None
 
 
+def upsum(m, dir_fd):
```

Comment:
Please include what `m` is in the docstring.

Should we put this into `upsum.py`? (I still like `treesum()`!)

This function needs a unit test.

Reply:
Updated the docstring. `treesum()` it is!
I agree on the unit test, though we don't do that at all at the moment, does it make sense to introduce unit tests for everything in a separate PR?

Reply:
I'll let it slide this time because I get the `treesum` name :tada: 

Seriously though: The fact that we don't have any tests shouldn't be a reason to add more code without tests. Things are moving fast still, so it's fine for now, but let's commit to a time in the (near) future from which point on we won't allow new code without a corresponding test. Probably when we add tests for the existing things (and this :wink:)

Reply:
Totally agreed. Just need to agree on how testing sholud be done, but that felt like a separate conversation.